integration-libs/opf/base/core/services/opf-pickup-in-store-handler.service.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, inject } from '@angular/core';
+import { ActiveCartFacade } from '@spartacus/cart/base/root';
+import { Product } from '@spartacus/core';
+import { OpfQuickBuyDeliveryType } from '@spartacus/opf/base/root';
+import { IntendedPickupLocationFacade } from '@spartacus/pickup-in-store/root';
+import { CurrentProductService } from '@spartacus/storefront';
+import { Observable, of } from 'rxjs';
+import { map, switchMap, take } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OpfPickupInStoreHandlerService {
+  protected currentProductService = inject(CurrentProductService);
+  protected activeCartFacade = inject(ActiveCartFacade);
+  protected intendedPickupLocationFacade = inject(IntendedPickupLocationFacade);
+
+  /**
+   * Retrieves the delivery type for a single product based on the intended pickup location.
+   * @return An observable emitting the delivery type (pickup or shipping).
+   */
+  getSingleProductDeliveryType(): Observable<OpfQuickBuyDeliveryType> {
+    if (!this.intendedPickupLocationFacade) {
+      return of(OpfQuickBuyDeliveryType.SHIPPING);
+    }
+
+    return this.currentProductService.getProduct().pipe(
+      take(1),
+      switchMap((product: Product | null) =>
+        this.intendedPickupLocationFacade
+          .getIntendedLocation(product?.code as string)
+          .pipe(
+            map((intendedLocation) => {
+              return intendedLocation?.pickupOption === 'pickup'
```

Comment:
might need to put 'pickup' in an enum. That will be catch by sonar.

Reply:
I guess `OpfQuickBuyDeliveryType.PICKUP.toLowerCase()` should do the trick. Pickup option enum is not exposed from pickup in store lib.